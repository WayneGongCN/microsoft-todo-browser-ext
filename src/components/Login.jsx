import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';

function Login(props) {
  const { login, loading } = props;
  return (
    <Button
      size="small"
      variant="contained"
      color="primary"
      disabled={loading}
      onClick={login}
      endIcon={loading && <CircularProgress size={20} />}
    >
      {loading ? 'Please waiting ...' : 'LOGIN'}
    </Button>
  );
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default Login;
