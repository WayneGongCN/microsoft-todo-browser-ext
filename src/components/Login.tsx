import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { State } from '../redux';
import { LANG_LOADING_TEXT, LANG_LOGIN_TEXT } from '../constants/lang';
import Button from '@mui/material/Button';
import CircularProgress from '@mui/material/CircularProgress';
import { loginAction } from '../redux/auth';



const Login: React.FC = () => {
  const dispatch = useDispatch();
  const loading = useSelector((state: State) => state.auth.loading);
  
  const handleLogin = useCallback(() => {
    dispatch(loginAction())
  }, [dispatch]);

  
  return (
    <Button
      fullWidth
      id="com-btn-login"
      color="primary"
      variant="contained"
      disabled={loading}
      endIcon={loading ? <CircularProgress size={20} /> : null}
      onClick={handleLogin}
    >
      {loading ? LANG_LOADING_TEXT + '...' : LANG_LOGIN_TEXT}
    </Button>
  );
};

export default Login;
