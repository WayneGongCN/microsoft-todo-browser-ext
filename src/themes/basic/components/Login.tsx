import { Button } from '@material-ui/core';
import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { State } from '../../../redux';
import { backgroundContext } from '../../../popup/';

const Login: React.FC<any> = () => {
  const dispatch = useDispatch();
  const account = useSelector((state: State) => state.auth.authenticationResult?.account);
  const loading = useSelector((state: State) => state.auth.loading);

  const handleLogin = useCallback(() => {
    dispatch(backgroundContext.authSlice.authentication());
  }, [dispatch]);

  useEffect(() => {
    if (!account) handleLogin();
  }, []);

  if (account) return <h2>Hello {account.name}</h2>;
  else if (loading) return <h2>Loading ...</h2>;
  else return <Button onClick={handleLogin}>Login</Button>;
};

export default Login;
