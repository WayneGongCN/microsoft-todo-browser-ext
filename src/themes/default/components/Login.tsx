import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { State } from '../../../redux';
import { LANG_POPUP_LOADING_TEXT, LANG_POPUP_LOGIN_TEXT } from '../../../constants/lang';
import Container from '@material-ui/core/Container';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import { backgroundContext } from '../..';

const { authSlice } = backgroundContext;

const Login: React.FC = () => {
  const dispatch = useDispatch();
  const loading = useSelector((state: State) => state.auth.loading);

  const handleLogin = useCallback(() => {
    dispatch(authSlice.authentication());
  }, [dispatch]);

  return (
    <Container style={{ margin: '12px 0' }}>
      <Button fullWidth color="primary" variant="contained" disabled={loading} endIcon={loading ? <CircularProgress size={20} /> : null} onClick={handleLogin}>
        {loading ? LANG_POPUP_LOADING_TEXT + '...' : LANG_POPUP_LOGIN_TEXT}
      </Button>
    </Container>
  );
};

export default Login;
