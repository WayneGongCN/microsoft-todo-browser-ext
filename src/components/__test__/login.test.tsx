import React from 'react';
import render from 'react-test-renderer'
import Login from "../Login";

it('Render', () => {
  const tree = render.create(<Login />).toJSON();

  expect(tree).toMatchSnapshot();
});