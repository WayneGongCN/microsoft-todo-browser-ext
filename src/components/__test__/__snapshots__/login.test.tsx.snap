// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test loading: loading 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
  disabled={true}
  id="com-btn-login"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  undefined...
  <span
    className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </span>
</button>
`;

exports[`Snapshot test normal: normal 1`] = `
<button
  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  id="com-btn-login"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
/>
`;
